pkg/cmd/server/bootstrappolicy/policy.go

```diff
@@ -213,6 +213,100 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 		},
 		{
 			ObjectMeta: kapi.ObjectMeta{
+				Name: BuildControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				// BuildControllerFactory.buildLW
+				// BuildControllerFactory.buildDeleteLW
+				{
+					Verbs:     util.NewStringSet("get", "list", "watch"),
+					Resources: util.NewStringSet("builds"),
+				},
+				// BuildController.BuildUpdater (OSClientBuildClient)
+				{
+					Verbs:     util.NewStringSet("update"),
+					Resources: util.NewStringSet("builds"),
+				},
+				// BuildController.ImageStreamClient (ControllerClient)
+				{
+					Verbs:     util.NewStringSet("get"),
+					Resources: util.NewStringSet("imagestreams"),
+				},
+				// BuildController.PodManager (ControllerClient)
+				// BuildDeleteController.PodManager (ControllerClient)
+				// BuildControllerFactory.buildDeleteLW
+				{
+					Verbs:     util.NewStringSet("get", "list", "create", "delete"),
+					Resources: util.NewStringSet("pods"),
+				},
+				// BuildController.Recorder (EventBroadcaster)
+				{
+					Verbs:     util.NewStringSet("create", "update"),
+					Resources: util.NewStringSet("events"),
+				},
+			},
+		},
+		{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: DeploymentControllerRoleName,
```

Comment:
@ironcladlou can you take a look at this list of permissions, and let me know if you see anything else the controller launched from RunDeploymentController() need? Note that the other RunDeploy*Controller functions are still using the highly-privileged client.